lms/djangoapps/discussion_api/api.py

```diff
@@ -672,8 +697,12 @@ def update_comment(request, comment_id, update_data):
     _check_editable_fields(cc_comment, update_data, context)
     serializer = CommentSerializer(cc_comment, data=update_data, partial=True, context=context)
     actions_form = CommentActionsForm(update_data)
-    if not (serializer.is_valid() and actions_form.is_valid()):
-        raise ValidationError(dict(serializer.errors.items() + actions_form.errors.items()))
+    if not actions_form.is_valid():
+        raise ValidationError(dict(actions_form.errors.items()))
+    if "voted" in update_data:
+        serializer = _update_vote_count(actions_form.cleaned_data["voted"], serializer)
+    if not serializer.is_valid():
+        raise ValidationError(dict(serializer.errors.items()))
```

Comment:
Consider refactoring this code since L700-705 are exactly the same as L653-658.  Perhaps expand the `_update_vote_count` method to include these additional lines.